documentation/public/user-guide/npm-package/how-to-parse-a-file/index.md

```diff
@@ -63,14 +63,32 @@ Let's do that:
 --8<-- "crates/solidity/outputs/npm/tests/src/doc-examples/reconstruct-source.ts:step-2-assertion"
 ```
 
-### Example 2: List the top-level contracts and their names
+### Example 2: Reading Contracts using Cursors
```

Comment:
Saying we "read" them in this example is, I feel like, a bit too generous - we're only fetching the name, which is a direct token child of the encompassing rule. It'd be better if we had some more examples on how to practically read the contracts (listing public/external functions, errors etc.) and use that wording there

Reply:
also I feel that "contracts" should be lower case since it refers to contracts generally; Cursor is okay since it's *the* Cursor API/structs we expose